frontend/src/index.tsx

```diff
@@ -3,10 +3,13 @@ import ReactDOM from "react-dom";
 import "./index.css";
 import App from "./App";
 import reportWebVitals from "./reportWebVitals";
+import ContextProviders from "./ContextProviders";
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <ContextProviders>
+      <App />
+    </ContextProviders>
```

Comment:
I wonder if it would be better to move the ContextProviders from the root to where we initialise the MemoryRouter for the game states since the context isn't needed or the home or nickname pages?

Reply:
Probably leave this how is for now. Perhaps we can look at changing it if we have time to research how this change might effect re-rendering